test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -168,6 +169,29 @@ public void signInWithEmail() throws Exception {
         account.setReauthToken(REAUTH_TOKEN);
         doReturn(account).when(accountDao).authenticate(study, EMAIL_PASSWORD_SIGN_IN);
         doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+        doReturn(CONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        
+        UserSession retrieved = service.signIn(study, CONTEXT, EMAIL_PASSWORD_SIGN_IN);
+        assertEquals(REAUTH_TOKEN, retrieved.getReauthToken());
+    }
+    
+    @Test(expected = ConsentRequiredException.class)
+    public void unconsentedSignInWithEmail() throws Exception {
+        doReturn(account).when(accountDao).authenticate(study, EMAIL_PASSWORD_SIGN_IN);
+        doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+        doReturn(UNCONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        
+        service.signIn(study, CONTEXT, EMAIL_PASSWORD_SIGN_IN);
+    }
+    
+    @Test
+    public void adminSignInWithEmail() throws Exception {
```

Comment:
might be nice to separate email and phone sign in as 2 tests.

Reply:
There are 3 tests for email+password and phone+password sign ins, one for a consented user, one for an uncontested user, and one for an uncontested administrative user, using both types of credentials. Are there additional tests you'd like to see here?

Reply:
No. My suggestion was to just separate them into multiple test methods. Not a big deal though.

Reply:
I think they are in multiple test methods. ?